### ISSUE TITLE
Stocktake: insert error explanation

### ISSUE THREAD

**Issue by mark-prins** [msupply-foundation/open-msupply#5297]
## Is your feature request related to a problem? Please describe 👀
Given this error: 
![telegram-cloud-photo-size-1-4990080244048309464-y](https://github.com/user-attachments/assets/323bd491-5f5d-4b8d-be22-25b10ff1bec9)

Can we have a way of seeing the actual batch, and a way to resolve the problem please?

See also #4035 and #3285

## Describe the solution you'd like 🎁
Instead of a toast notification which disappears quickly and can only display limited information, could we show a modal and have links to the affected batches?

Currently it is hard to know how to fix the issue. Having more information would be useful - at the least, display the name of the item and the batch. Ideally it should have a link to the batch, so that you can edit the hold status.

EDIT: Use existing functionality for highlighting which lines are erroring as well

**Comment by GaryWilletts** [comment]
@mark-prins why is there a check if a line is on hold? Stocktakes don't care, you're just doing the count, same with inventory adjustments. Other outbound transactions do care but not stocktakes or inventory adjustments I thought. This is true in OG anyway

**Comment by andreievg** [comment]
Hmm we already have a mechanism for displaying errors, but there is some ambiguity about that mechanism. I set needs refinement and high priority for one of the issues that @mark-prins linked. For this one, for now we can ignore this error for stocktake, on backend. Also added needs refinement and high

**Comment by andreievg** [comment]
After refinement, the proposed solution is two part (first higher priority then the second):

1 - allow on hold stocklines to be adjustable in stocktakes.
2 - error to be mapped or displayed correctly, would require more structured error [here](https://github.com/msupply-foundation/open-msupply/blob/48255c1eea9063dc3a288fe9d11dee6b0299a9f2/server/graphql/stocktake/src/mutations/update.rs#L150-L151) so it doesn't throw an error [here](https://github.com/msupply-foundation/open-msupply/blob/48255c1eea9063dc3a288fe9d11dee6b0299a9f2/client/packages/inventory/src/Stocktake/DetailView/Footer/StatusChangeButton.tsx#L148-L149)

**Comment by andreievg** [comment]
Also added another issue here: https://github.com/msupply-foundation/open-msupply/issues/5363

**Comment by louisaw123** [comment]
Yes we just had this problem occur in timor. There was no way for the team to find the affected stock line. We have had to do this from msupply :(

**Comment by mark-prins** [comment]
triage: many comments. recommend a cooldown task given how full the next release is

**Comment by louisaw123** [comment]
Hi @msupply-foundation/triage . Are there any plans for this one in the immediate term still? I've been getting some more issues raised by client and it's not really a good look that this is still occuring for them and they cannot find the affected item and are not able to complete their stocktake which really should not be sitting unfinalised for long periods of time.. especially since they have counted 200 items and are unable to work out which one is the issue.

<img width="441" height="151" alt="Image" src="https://github.com/user-attachments/assets/bb1fa78c-225a-480f-a023-60df5b6f9ad8" />

**Comment by andreievg** [comment]
Linked issue for measure of impact/importance [#10133](https://github.com/msupply-foundation/open-msupply/issues/10133)
